# Kikilicious Data Analysis


### Context

Food delivery platforms generate large volumes of transactional data that can be leveraged to understand customer behaviour, restaurant performance, and operational efficiency. The dataset of Kikilicious Foodhub captures orders, customer interactions, delivery times, and ratings within a food delivery ecosystem.

### Objective
To analyse Kikilicious FoodHub’s operational and customer data in order to:
1. Identify patterns in customer ordering behaviour
2. Evaluate restaurant performance and customer satisfaction
3. Assess delivery efficiency and its impact on user ratings
4. Generate data-driven insights that can improve service quality, reduce delays, and enhance customer retention

### Data Description
The data contains the different data related to a food order. The detailed data dictionary is given below.

### Data Dictionary

* order_id: Unique ID of the order
* customer_id: ID of the customer who ordered the food
* restaurant_name: Name of the restaurant
* cuisine_type: Cuisine ordered by the customer
* cost: Cost of the order
* day_of_the_week: Indicates whether the order is placed on a weekday or weekend (The weekday is from Monday to Friday and the weekend is Saturday and Sunday)
* rating: Rating given by the customer out of 5
* food_preparation_time: Time (in minutes) taken by the restaurant to prepare the food. This is calculated by taking the difference between the timestamps of the restaurant's order confirmation and the delivery person's pick-up confirmation.
* delivery_time: Time (in minutes) taken by the delivery person to deliver the food package. This is calculated by taking the difference between the timestamps of the delivery person's pick-up confirmation and drop-off information

## Import the Required Libraries

In [2]:
import numpy as np
import pandas as pd
import matplotlib
import matplotlib.pyplot as plt
import matplotlib.style as style
import seaborn as sns
%matplotlib inline

## Import Data

In [3]:
df = pd.read_csv("kikilicious_orders.csv")

## Sanity Checks on the Data

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1898 entries, 0 to 1897
Data columns (total 9 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   order_id               1898 non-null   int64  
 1   customer_id            1898 non-null   int64  
 2   restaurant_name        1898 non-null   object 
 3   cuisine_type           1898 non-null   object 
 4   cost_of_the_order      1898 non-null   float64
 5   day_of_the_week        1898 non-null   object 
 6   rating                 1898 non-null   object 
 7   food_preparation_time  1898 non-null   int64  
 8   delivery_time          1898 non-null   int64  
dtypes: float64(1), int64(4), object(4)
memory usage: 133.6+ KB


In [6]:
print(f"Kikilicious dataset has {df.shape[0]} rows and {df.shape[1]} columns.")

Kikilicious dataset has 1898 rows and 9 columns.


In [7]:
# First five rows of the dataset
df.head()

,order_id,customer_id,restaurant_name,cuisine_type,cost_of_the_order,day_of_the_week,rating,food_preparation_time,delivery_time
0,1477147,337525,Hangawi,Korean,30.75,Weekend,Not given,25,20
1,1477685,358141,Blue Ribbon Sushi Izakaya,Japanese,12.08,Weekend,Not given,25,23
2,1477070,66393,Cafe Habana,Mexican,12.23,Weekday,5,23,28
3,1477334,106968,Blue Ribbon Fried Chicken,American,29.20,Weekend,3,25,15
4,1478249,76942,Dirty Bird to Go,American,11.59,Weekday,4,25,24


In [8]:
#Last five rows of the dataset
df.tail()

,order_id,customer_id,restaurant_name,cuisine_type,cost_of_the_order,day_of_the_week,rating,food_preparation_time,delivery_time
1893,1476701,292602,Chipotle Mexican Grill $1.99 Delivery,Mexican,22.31,Weekend,5,31,17
1894,1477421,397537,The Smile,American,12.18,Weekend,5,31,19
1895,1477819,35309,Blue Ribbon Sushi,Japanese,25.22,Weekday,Not given,31,24
1896,1477513,64151,Jack's Wife Freda,Mediterranean,12.18,Weekday,5,23,31
1897,1478056,120353,Blue Ribbon Sushi,Japanese,19.45,Weekend,Not given,28,24


Both the first five and last five rows of the dataset give a general overview or a sneak peek into how what the dataset loooks like, the column names and what kind of analysis to expect.

In [9]:
df.describe()

,order_id,customer_id,cost_of_the_order,food_preparation_time,delivery_time
count,1.898000e+03,1898.000000,1898.000000,1898.000000,1898.000000
mean,1.477496e+06,171168.478398,16.498851,27.371970,24.161749
std,5.480497e+02,113698.139743,7.483812,4.632481,4.972637
min,1.476547e+06,1311.000000,4.470000,20.000000,15.000000
25%,1.477021e+06,77787.750000,12.080000,23.000000,20.000000
50%,1.477496e+06,128600.000000,14.140000,27.000000,25.000000
75%,1.477970e+06,270525.000000,22.297500,31.000000,28.000000
max,1.478444e+06,405334.000000,35.410000,35.000000,33.000000


Kikilicious catered to 1898 orders. The averages of the cost of orders, food preparation time as well the delivery time are summarised here.

In [10]:
df.describe(include = "all")

,order_id,customer_id,restaurant_name,cuisine_type,cost_of_the_order,day_of_the_week,rating,food_preparation_time,delivery_time
count,1.898000e+03,1898.000000,1898,1898,1898.000000,1898,1898,1898.000000,1898.000000
unique,NaN,NaN,178,14,NaN,2,4,NaN,NaN
top,NaN,NaN,Shake Shack,American,NaN,Weekend,Not given,NaN,NaN
freq,NaN,NaN,219,584,NaN,1351,736,NaN,NaN
mean,1.477496e+06,171168.478398,NaN,NaN,16.498851,NaN,NaN,27.371970,24.161749
std,5.480497e+02,113698.139743,NaN,NaN,7.483812,NaN,NaN,4.632481,4.972637
min,1.476547e+06,1311.000000,NaN,NaN,4.470000,NaN,NaN,20.000000,15.000000
25%,1.477021e+06,77787.750000,NaN,NaN,12.080000,NaN,NaN,23.000000,20.000000
50%,1.477496e+06,128600.000000,NaN,NaN,14.140000,NaN,NaN,27.000000,25.000000
75%,1.477970e+06,270525.000000,NaN,NaN,22.297500,NaN,NaN,31.000000,28.000000


In [11]:
df.isnull().sum()

order_id                 0
customer_id              0
restaurant_name          0
cuisine_type             0
cost_of_the_order        0
day_of_the_week          0
rating                   0
food_preparation_time    0
delivery_time            0
dtype: int64

This shows that the dataset has no missing values.

In [12]:
#Checking for datatypes
df.dtypes

order_id                   int64
customer_id                int64
restaurant_name           object
cuisine_type              object
cost_of_the_order        float64
day_of_the_week           object
rating                    object
food_preparation_time      int64
delivery_time              int64
dtype: object

The dataset has three different datatypes consisting of categorical and numerical(whole numbers and decimals).

## Orders not Rated

In [13]:
df[["rating"]].value_counts()[["Not given"]]

rating   
Not given    736
Name: count, dtype: int64

736 orders were not rated at all.

## Exploratory Data Analysis (EDA)

### Univariate Analysis

### Variable Distributions

### Top 5 restauraurants by orders received

### Most popular weekend cuisine

### Percentage of the orders that cost more than 20 dollars

### Average delivery time of orders

### IDs of the top 3 customers and the number of orders they placed

## Bivariate/Multivariate Analysis

### Bivariate/multivariate analysis exploring relationships between the important variables in the dataset


### **Question 9:** Promotional offer based on rating counts of restaurants working with Kikilicious Foodhub

### Net revenue based on order cost

### Total time required to deliver the food. Percentage of orders that take more than 60 minutes to get delivered from the time the order is placed

### Differences in the delivery time of the orders on weekdays and weekends

## Conclusion and Recommendations

### **Conclusions from the analysis.** Recommendation to help improve the business based on cuisine type and feedback ratings

### Conclusions:
*  

### Recommendations:

*  

---